lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -354,7 +354,7 @@ def _connect_uncached(self):
         except paramiko.ssh_exception.BadHostKeyException as e:
             raise AnsibleConnectionFailure('host key mismatch for %s' % e.hostname)
         except Exception as e:
-            msg = str(e)
+            msg = to_text(e)
```

Comment:
It would be safer (but not required) to change the literal strings that are being compared to msg into u"" strings.  The reason is that while ```in``` will do the correct thing if the literal string is entirely ascii, if we ever use non-ascii in the literal string or we use ```==``` instead of in, the comparison will fail on Python2 (traceback in the first place, always return False in the second case).  Changing it now isn't necessary but it will hopefully remind anyone working on this in the future that they need to be careful of that.